src/openms_gui/source/VISUAL/APPLICATIONS/TOPPViewBase.cpp

```diff
@@ -732,12 +732,23 @@ namespace OpenMS
         OPENMS_LOG_INFO << "INFO: done loading all " << std::endl;
 
         // a mzML file may contain both, chromatogram and peak data
-        // -> this is handled in PlotCanvas::addPeakLayer
-        data_type = LayerDataBase::DT_CHROMATOGRAM;
-        if (peak_map_sptr->containsScanOfLevel(1))
+        // -> this is handled in PlotCanvas::addPeakLayer FIXME: No it's not!
+        if (peak_map_sptr->getNrChromatograms() > 0)
         {
+          data_type = LayerDataBase::DT_CHROMATOGRAM;
+        }
+        else if (peak_map_sptr->getNrSpectra() > 0)
+        {
+          if (data_type == LayerDataBase::DT_CHROMATOGRAM)
```

Comment:
this cannot possibly happen, right?

Reply:
I think it can't 
@poshul data_type is initialized as LayerDataBase::DT_UNKNOWN and only set to LayerDataBase::DT_CHROMATOGRAM in line 738 - the other if branch

Reply:
ah, I see, I've updated the flow to correct this

Reply:
there is a less fragile way of fixing it:
explicitly compare for peak_map_sptr->getNrChromatograms() > 0 && peak_map_sptr->getNrSpectra() > 0

note that your code now depends on the lines above which should be prevented if easily possible.